smart-cart/app/controllers/recommendations_controller.rb

```diff
@@ -1,18 +1,51 @@
 class RecommendationsController < ApplicationController
+    
+
     def show
-        require 'json'
-        l = ['milk', 'eggs']
-        result = `python3 #{ENV["PWD"] + "/app/controllers/print_dict.py"} "'#{l}'"` # pass l as an argument 
-        $hash = JSON.parse(result.gsub("'", "\"")) # turn string result into a hash
-        @hello = "hello"
-        print $hash
-        $stores = [$hash['1']['store'], $hash['2']['store'], $hash['3']['store']]
+        session[:current_user_id] = 1234567
+        session_id = session[:current_user_id]
+
+        # check if there is already an entry in Recommendations for the current user
+        # if not, then we will save the recommendations generated by search_v3.py to Recommendations table
+        if (Recommendation.find_by(list_id: session_id).blank?)
+            require 'json'
+            l = ['milk', 'eggs']
+            result = `python3 #{ENV["PWD"] + "/app/controllers/print_dict.py"} "'#{l}'"` # pass l as an argument 
```

Comment:
when you run the search script eventually add `-W ignore` to ignore some weird warnings 
 `python3 -W ignore ...`

this is just an fyi for the future